src/features/vars-schema/env-vars.ts

```diff
@@ -56,7 +56,14 @@ export const INVALID_NEWLINE_REGEX = /\r|\n/;
 // Note that this list will be mutated by the cloud API to add
 // extra variables as for cloud-only features.
 export const SUPERVISOR_CONFIG_VAR_PROPERTIES: {
-	[k: string]: JSONSchema6Definition;
+	[k: string]:
+		| JSONSchema6Definition
+		| {
+				properties: {
+					will_reboot?: boolean;
+					warning?: string;
+				};
+		  };
```

Comment:
```suggestion
		JSONSchema6Definition & {
				properties?: {
					will_reboot?: boolean;
					warning?: string;
				};
		  };
```

Reply:
Thanks @dfunckt 

> | JSONSchema6Definition

Good catch on this, this is incorrect but I used a bitwise OR (`|`)(https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_OR) here instead of a bitwise AND (`&`) because I feel we are looking for a union of types (addition) and using a `&` made vscode mad. So I used the single pipe `|`

The error I was getting when hovering over line 68.

```
Type '{ enum: string[]; description: string; default: string; }' is not assignable to type 'JSONSchema6Definition & { properties: { will_reboot?: boolean | undefined; warning?: string | undefined; }; }'.
  Type '{ enum: string[]; description: string; default: string; }' is not assignable to type 'JSONSchema6 & { properties: { will_reboot?: boolean | undefined; warning?: string | undefined; }; }'.
    Property 'properties' is missing in type '{ enum: string[]; description: string; default: string; }' but required in type '{ properties: { will_reboot?: boolean | undefined; warning?: string | undefined; }; }'.ts(2322)
```

![Screenshot from 2021-12-13 19-13-01](https://user-images.githubusercontent.com/22801822/145823154-a6679471-96c8-4263-a7e7-167ef75b0556.png)

Reply:
These operators work like *logical* AND/OR, not bitwise @vipulgupta2048. Counter-intuitive, right? `&` creates the union of two types. `|` on the other hand for, say, `A | B` is "A *or* B".

The reason you get all these errors with `&` is because for most vars you don't have the `properties` attribute and it is declared as non-optional. To fix this you just need to declare it as `properties?: { ... }` (note the question mark) which also semantically better -- i.e. it's existence is optional and introduces new behaviour beyond the default.

Reply:
> logical AND/OR

I tried to follow the docs on this one instead of what was actually happening. Hence, ended up choosing the single pipe `|`.  

Good touch on the optional properties object @dfunckt on my first pass, I marked properties of the `properties` object as optional. But this is way cleaner.